### PR TITLE
Produce a fake `BeamSpot` object in `BeamSpotOnlineProducer` when using transient record logic and `OnlineBeamSpotESProducer` returned a fake

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

While working on https://github.com/cms-sw/cmssw/pull/41193/, I have noticed that despite `OnlineBeamSpotESProducer`  has a refined logic in order to put into the event setup a fake BeamSpot when certain conditions are not met:

https://github.com/cms-sw/cmssw/blob/a52f2d1ecf97bde0e9f10d45959c88337c481687/RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc#L168-L174

the current `BeamSpotOnlineProducer` logic, if `useTransientRecord_` is true and the  `OnlineBeamSpotESProducer` returned a fake beamspot, just falls back to DB (reading the content of `BeamSpotObjectsRcd`, populated by the PCL in realtime workflows)  - which I have understood from the Beam Spot experts (@gennai) is not the right / expected behaviour.

https://github.com/cms-sw/cmssw/blob/a52f2d1ecf97bde0e9f10d45959c88337c481687/RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc#L104-L111

In this PR (in commit ec786fe40171051cc0b63074795fe6dedeeed2eb) I change the current behaviour, such that instead now it will indeed produce a fake BS for consumption online (as it was originally devised).
This has a consequence, that in the case in which we want indeed to fall-back to DB (e.g. for the Phase-2 HLT testing case, after https://github.com/cms-sw/cmssw/pull/41193), the change will alter physics in an undesired way. For this reason, it has been necessary to supply to the `auto:phase2_realistic` GlobalTag key the right  `BeamSpotOnline*ObjectsRcd` payloads in order for the  `OnlineBeamSpotESProducer` to not pick up a fake BeamSpot. This is done in commit  5fbea8ac08aea2e828674374d12c50f7b254d9d9, while commit 30121dd26d054de4f81106233278183baf9da685 sets the the time threshold to `1e6` seconds (as it's done for the Run-3 menus, see https://github.com/cms-sw/cmssw/pull/41193/files#r1152358315) to the same effect.
The difference in Global Tags is:
   * https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/131X_mcRun4_realistic_v4/131X_mcRun4_realistic_v5

To supply such payloads, I have found convenient to create a new plugin `BeamSpotOnlineFromOfflineConverter` that taking a `BeamSpotObject` in input creates an sqlite file in output containing a payload of the type `BeamSpotOnlineObjects`. This has been added in commit e870b172a627b65f531f710e41aecacef1d12c7c, while commits a03a99d5a5bb21ea5bdf1ef852b8845d336f4f34 and a03a99d5a5bb21ea5bdf1ef852b8845d336f4f34 add a utility accessor to the `BeamSpotOnlineObjects` condtion format to be able to readily copy from a `BeamSpotObjects` payload (friend class) all the common parameters. Finally commit ecd817c8349e160a65930c6da4eaf5248419dc98 adds this converter to the battery of unit tests of the `CondTools/BeamSpot` package. 

#### PR validation:

`cmssw` complies. Passes back unit tests.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

To be backported down to 13.0.X for 2023 data-taking purposes

@francescobrivio @dzuolo

### CODE REVIEW

File: Configuration/AlCa/python/autoCond.py [hunk 1]
```diff
@@ -86,7 +86,7 @@
     # GlobalTag for MC production with realistic conditions for Phase1 2024
     'phase1_2024_realistic'        : '131X_mcRun3_2024_realistic_v3',
     # GlobalTag for MC production with realistic conditions for Phase2
-    'phase2_realistic'             : '131X_mcRun4_realistic_v4'
+    'phase2_realistic'             : '131X_mcRun4_realistic_Candidate_2023_05_06_19_12_47'
```
Comment (mmusich) [inline Configuration/AlCa/python/autoCond.py:89]: @cms-sw/alca-l2 if this PR is agreed upon, we'll need a full-fledged GloblaTag implementing this.

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Configuration/AlCa/python/autoCond.py (hunk 1 above)
Comment (francescobrivio) [reply Configuration/AlCa/python/autoCond.py:89]: Ciao @mmusich thanks for this!
In general it looks good, give me just a bit more of time to review the updated logic and we will provide a versioned GT based on your candidate.

File: HLTrigger/Configuration/python/HLT_75e33/eventsetup/hltOnlineBeamSpotESProducer_cfi.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import FWCore.ParameterSet.Config as cms
 
 hltOnlineBeamSpotESProducer = cms.ESProducer("OnlineBeamSpotESProducer",
-    timeThreshold = cms.int32( 48 ),
+    timeThreshold = cms.int32( 1e6 ),   # we do want to read the DB even if it's old
```
Comment (mmusich) [inline HLTrigger/Configuration/python/HLT_75e33/eventsetup/hltOnlineBeamSpotESProducer_cfi.py:4]: mmh, 
> ValueError: 1000000.0 is not a valid <class 'FWCore.ParameterSet.Types.int32'>

Comment (francescobrivio) [reply HLTrigger/Configuration/python/HLT_75e33/eventsetup/hltOnlineBeamSpotESProducer_cfi.py:4]: maybe try `999999`? as in:
https://github.com/cms-sw/cmssw/blob/a52f2d1ecf97bde0e9f10d45959c88337c481687/Configuration/DataProcessing/python/RecoTLR.py#LL10C25-L10C31

Comment (mmusich) [reply HLTrigger/Configuration/python/HLT_75e33/eventsetup/hltOnlineBeamSpotESProducer_cfi.py:4]: the HLT customization has `int(1e6)`:

https://github.com/cms-sw/cmssw/blob/fa01ae41a33c9ecaa9771e2259c455a6e942f068/HLTrigger/Configuration/python/customizeHLTforCMSSW.py#LL209C30-L210C1

I overlooked it yesterday.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
> DQMHistoTests: Total failures: 59

so out of the (more relevant) failures in the [DQM bin-by-bin comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_2_X_2023-05-07-2300+54ccf3/56916/dqm-histo-comparison-summary.html)
   * [138.5](https://cmsweb.cern.ch/dqm/dev/session/TlmbuB) looks spurious (?)
   * [1000.0](https://tinyurl.com/2lnmkc3d) looks spurious
   * [138.4](https://tinyurl.com/2eygykty) is expected, because it's using `auto:run3_data_prompt` and that GT doesn't have the online BeamSpot records (see https://github.com/cms-sw/cmssw/issues/41459) and therefore the `ESProducer` is generating a fake beamspot and the `BeamSpotOnlineProducer` now picks up the fake value instead of falling back DB
   * [139.001](https://tinyurl.com/2g3zz2g9) is expected, because it's using `auto:run3_data_relval` (same reasons as above).

**Comment by francescobrivio** [comment]
@mmusich 
Ok we made the new GT: `131X_mcRun4_realistic_v5`
 - please add to the PR description the diff link: https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/131X_mcRun4_realistic_v4/131X_mcRun4_realistic_v5
 - for reference, I have checked the new BeamSpotOnline tag and verified that it's compatible with `BeamSpotObjects_HLLHC_z4p3cm_t193ns_14TeV_nominal_mc`
   - the PayloadInspector plot (https://cern.ch/go/xd87) for the new tag is:
BSOnline_PhaseII

**Comment by francescobrivio** [comment]
@mmusich the last force push bringed in commit e14a94b3248d31c3ae72ebc3271f368746ec896b which I dont' think belongs in this PR?

**Comment by mmusich** [comment]
>the last force push bringed in commit https://github.com/cms-sw/cmssw/commit/e14a94b3248d31c3ae72ebc3271f368746ec896b which I dont' think belongs in this PR?

for some reason, which is frankly beyond me, gitHub doesn't allow me to update the remote of this branch anymore. I am going to close it and re-open it.

**Comment by iarspider** [comment]
> > the last force push bringed in commit [e14a94b](https://github.com/cms-sw/cmssw/commit/e14a94b3248d31c3ae72ebc3271f368746ec896b) which I dont' think belongs in this PR?
> 
> for some reason, which is frankly beyond me, gitHub doesn't allow me to update the remote of this branch anymore. I am going to close it and re-open it.

https://www.githubstatus.com/ 
TL;DR Most of Github functionality is broken

**Comment by mmusich** [comment]
>  TL;DR Most of Github functionality is broken

ah, too bad :) I thought I messed up something. Anyway here's the new one: https://github.com/cms-sw/cmssw/pull/41597
